superset/assets/cypress/integration/dashboard/dashboard_controls.js

```diff
@@ -0,0 +1,52 @@
+import { WORLD_HEALTH_DASHBOARD, getDashboardUnfavorEndpoing } from './dashboard.helper';
+
+describe('dashboard top-level controls', () => {
+  let sliceIds = [];
+  let dashboard = {};
+
+  beforeEach(() => {
+    cy.server();
+    cy.login();
+
+    cy.visit(WORLD_HEALTH_DASHBOARD);
+
+    cy.get('#app').then((data) => {
+      const bootstrapData = JSON.parse(data[0].dataset.bootstrap);
+      dashboard = bootstrapData.dashboard_data;
+      sliceIds = dashboard.slices.map(slice => (slice.slice_id));
+    });
+  });
+
+  it('should allow favor/unfavor', () => {
+    // make sure dashboard is not favorite
+    cy.request(getDashboardUnfavorEndpoing(dashboard.id));
+    cy.visit(WORLD_HEALTH_DASHBOARD);
```

Comment:
It looks like the visit is happening in the `beforeEach`, is there a reason to also have it here?

Reply:
in beforeEach, get dashboard_id. Inside test, first call backend to enforce dashboard not favorite. Then load dashboard again, so that at this moment i can assume the initial state is unfavored. 

But now i think the test can write in another way: depends on the favorite state from beforeEach, i can test star UI first and then toggle it.